### ISSUE TITLE
LCOV report type empty when using C# code coverage JSON file

### ISSUE THREAD

**Issue by CalvinWilkinson** [danielpalme/ReportGenerator#435]
**Describe the bug**
I am currently having an issue where the produced **lcov.info** file is empty.  The only content it has in it is a single line of text at the top that is **TN**.

I have a **NET 5.0** library that I am running tests for using **GitHub actions**.  Once the tests produce the coverage result in **lcov** format, I am pointing the **reportgenerator** to the produced **coverage.json** file created with the tests from the previous step.

I am trying to use the **coverallsapp v1.1.2** GitHub action to pick up the **lcov.info** file and that is when **coveralls** reports back to me that it cannot parse the string.  Upon investigation, revealed that the **lcov.info** file was empty.

**coveralls error:**
![image](https://user-images.githubusercontent.com/85414302/124112180-06ada880-da38-11eb-83a6-6f7321be6562.png)

**This is the unit testing section of my YAML**
``` yaml
  Run_Unit_Tests:
    runs-on: windows-latest
    steps:
      - uses: actions/checkout@v2
      
      - name: Setup .NET 5.0.x
        uses: actions/setup-dotnet@v1
        with:
          dotnet-version: 5.0.x
      
      - name: Restore Dependencies
        run: dotnet restore
      
      - name: Execute Unit Tests
        run: dotnet test "${{ github.workspace }}\\Testing\\CASLTests\\CASLTests.csproj" -c Debug-WIN /p:CollectCoverage=true /p:CoverletOutput=TestResults/ /p:CoverletOuputFormat=lcov
      
      - name: ReportGenerator
        uses: danielpalme/ReportGenerator-GitHub-Action@4.8.11
        with:
          reports: "**/TestResults/coverage.json"
          targetdir: "${{ github.workspace }}\\Testing\\CASLTests\\TestResults"
          reporttypes: lcov

      - name: Publish Code Coverage Results
        uses: coverallsapp/github-action@v1.1.2
        with:
          github-token: ${{ secrets.COVERALLS_TOKEN }}
          path-to-lcov: "${{ github.workspace }}\\Testing\\CASLTests\\TestResults\\lcov.info"
```

**To Reproduce**
1. Console output of ReportGenerator
![image](https://user-images.githubusercontent.com/85414302/124112621-791e8880-da38-11eb-9081-d0a91801e089.png)

2. Here is a zip with 2 files in it.  The **coverage.json** files is the file produced by the **dotnet cli**.  Of course the **lcov.info** file is produced by the report generator.
[GeneratedCoverageFiles.zip](https://github.com/danielpalme/ReportGenerator/files/6747575/GeneratedCoverageFiles.zip)

To reproduce this, I simply ran the same commands that is being performed in the **YAML** file.

**Comment by danielpalme** [comment]
ReportGenerator can't handle the JSON format generated by coverlet, you can use _OpenCover_ or _Coberatura_ format instead. 
See: https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/GlobalTool.md#coverage-output  

If you adjust your action, it should work:

```yaml
      - name: Execute Unit Tests
        run: dotnet test "${{ github.workspace }}\\Testing\\CASLTests\\CASLTests.csproj" -c Debug-WIN /p:CollectCoverage=true /p:CoverletOutput=TestResults/ /p:CoverletOuputFormat=cobertura
      
      - name: ReportGenerator
        uses: danielpalme/ReportGenerator-GitHub-Action@4.8.11
        with:
          reports: "**/TestResults/coverage.cobertura.xml"
          targetdir: "${{ github.workspace }}\\Testing\\CASLTests\\TestResults"
          reporttypes: lcov
```

**Comment by CalvinWilkinson** [comment]
Ok np.  

I will try this out at see what I get.  But one thing I am wondering is your comment about not handling JSON format.

When I use the command below, even though I am being explicit that the format is **Cobertura**, the file produces still shows up as a JSON file. So I feel this is not gonna work.

**Command Used:**
```dotnet test "MyProjectcsproj" -c Debug-WIN /p:CollectCoverage=true /p:CoverletOutput=TestResults/ /p:CoverletOuputFormat=cobertura```

**Comment by CalvinWilkinson** [comment]
So I did what you recommended and I indeed produced a **Cobertura.xml** file, but it is empty.

The command above did produce a file, but it was a file named **coverage.json** which is not empty.  

Any recommendations on how to proceed further?

**Comment by CalvinWilkinson** [comment]
Quick Update

So I was able to get the **Cobertura.xml** to not be empty by using a different command.  I will admit that I do not understand why one works over the other and the reason or true difference between the 2, but it is working.

**Old Command:**
```dotnet test "MyProject.csproj" -c Debug /p:CollectCoverage=true /p:CoverletOutput=TestResults/ /p:CoverletOuputFormat=cobertura```

**New Command:**
```dotnet test "MyProject.csproj" -c Debug --collect:"XPlat Code Coverage"```

This means that I did not need to use **ReportGenerator** at all.  The new command simply generated the **coverage.cobertura.xml** file for me, which I had picked up by **codecov**.

I am good to go.  Thanks for your help!!  It led me in the right direction to solve the problem.

I just wanted to update this issue to hopefully help other people out.
